### ISSUE TITLE
[Issue]: DLNA Shows Empty Folder Even though It is not empty.

### ISSUE THREAD

**Issue by SandyRodgers-2017** [jellyfin/jellyfin#7990]
### Please describe your bug

When I have DLNA enabled and access it, Jellyfin shows an empty folder for one of my libraries even though it isn't empty. In particular my movies directory and some of my genre folder. However, my other folders seemingly work correctly.  When I copy  over the same files to another directory, make a new library, and try to use it. The same problem persists.

### Jellyfin Version

10.8.0

### Environment

```markdown
- OS:Linux Mint 20 
- Virtualization:
- Clients:Browser, Android, VLC, Jellyfin Media Player
- Browser:Firefox, Chrome, Brave
- FFmpeg Version: jellyfin-ffmpeg5
- Playback Method: Direct Play,Remux
- Hardware Acceleration: VAAPI
- Plugins:Fanart, AniDB, AniList, AuidoDB, CoverArtARchive,Intro Skipper, Jellyfin Jav, Kodi Sync Queue, MusicBrainz, OMDb, Open Subtitles, Playback Reporter, Reports, Session Cleaner, Skin Manager, Studio Images, TMDb, TMDb Trailer, TVmaze TheTVDB, Theme Songs, YoutubeMetadata
- Reverse Proxy:none
- Base URL: None
- Networking:Host
- Storage:local
```

### Jellyfin logs

```shell
[2022-06-20 19:56:20.605 -04:00] [ERR] Error processing control request
System.NullReferenceException: Object reference not set to an instance of an object.
   at MediaBrowser.Model.Dlna.StreamBuilder.BuildVideoItem(MediaSourceInfo item, VideoOptions options)
   at MediaBrowser.Model.Dlna.StreamBuilder.BuildVideoItem(VideoOptions options)
   at Emby.Dlna.Didl.DidlBuilder.AddVideoResource(XmlWriter writer, BaseItem video, String deviceId, Filter filter, StreamInfo streamInfo)
   at Emby.Dlna.Didl.DidlBuilder.WriteItemElement(XmlWriter writer, BaseItem item, User user, BaseItem context, Nullable`1 contextStubType, String deviceId, Filter filter, StreamInfo streamInfo)
   at Emby.Dlna.ContentDirectory.ControlHandler.HandleBrowse(XmlWriter xmlWriter, IReadOnlyDictionary`2 sparams, String deviceId)
   at Emby.Dlna.ContentDirectory.ControlHandler.WriteResult(String methodName, IReadOnlyDictionary`2 methodParams, XmlWriter xmlWriter)
   at Emby.Dlna.Service.BaseControlHandler.CreateControlResponse(ControlRequestInfo requestInfo)
   at Emby.Dlna.Service.BaseControlHandler.ProcessControlRequestInternalAsync(ControlRequest request)
   at Emby.Dlna.Service.BaseControlHandler.ProcessControlRequestAsync(ControlRequest request)
[2022-06-20 19:56:20.605 -04:00] [WRN] Slow HTTP Response from "http://192.168.1.24:8096/dlna/2fcccd0b-684f-4834-902a-7389689b994f/contentdirectory/control" to "192.168.1.23" in 0:00:00.6777933 with Status Code 200
```

### Please attach any screenshots here

![genre_action](https://user-images.githubusercontent.com/101082169/174871221-df41b0d9-cda4-4172-8d78-f75b6483bcb3.jpg)
![genres](https://user-images.githubusercontent.com/101082169/174871236-b1270bcf-34af-49fb-97f9-7d637e159eaf.jpg)
![movies_directory](https://user-images.githubusercontent.com/101082169/174871248-7da67c50-a9c6-4c64-96ae-463603f1ce8c.jpg)
![Movies_Library_UPnP](https://user-images.githubusercontent.com/101082169/174871271-d3043385-816f-4376-b172-8bebae9ad4e8.jpg)
![Movies_Library_Deeper_UPnP](https://user-images.githubusercontent.com/101082169/174871291-aceb6dee-aa38-46e0-b959-d7d2347edc12.jpg)
![Genre_Action_UPnP](https://user-images.githubusercontent.com/101082169/174871309-4ce13c1e-5fe2-4e9f-a0d0-53c249fb823a.jpg)
![Anime_Library_UpnP](https://user-images.githubusercontent.com/101082169/174871331-a3c4b0b4-8fb3-49ce-9481-067acb07530c.jpg)
![Anime_Library_UpnP_Deeper](https://user-images.githubusercontent.com/101082169/174871359-33566a6f-ac65-4d32-b2a6-a26ad6fdc141.jpg)
![Anime_Library_UpnP_Deeper_AOT ](https://user-images.githubusercontent.com/101082169/174871372-64ed557b-8066-4ce6-89c3-3cc8436c4547.jpg)
![Anime_Library_UpnP_Deeper_AOT_S01](https://user-images.githubusercontent.com/101082169/174871393-4eabfc8a-8b65-44f6-9ae5-a8d787707eb5.jpg)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by SandyRodgers-2017** [comment]
fixed by 10.8.1

**Comment by SandyRodgers-2017** [comment]
fixed by upgrade to 10.8.1

**Comment by majid-vaghari** [comment]
I have this problem on ArchLinux using version 10.8.7
I installed it using "Generic Linux" tarball.

```[2022-11-14] [ERR] [46] Emby.Dlna.ContentDirectory.ContentDirectoryService: Error processing control request
System.NullReferenceException: Object reference not set to an instance of an object.
   at MediaBrowser.Model.Dlna.StreamBuilder.BuildVideoItem(MediaSourceInfo item, VideoOptions options)
   at MediaBrowser.Model.Dlna.StreamBuilder.BuildVideoItem(VideoOptions options)
   at Emby.Dlna.Didl.DidlBuilder.AddVideoResource(XmlWriter writer, BaseItem video, String deviceId, Filter filter, StreamInfo streamInfo)
   at Emby.Dlna.Didl.DidlBuilder.WriteItemElement(XmlWriter writer, BaseItem item, User user, BaseItem context, Nullable`1 contextStubType, String deviceId, Filter filter, StreamInfo streamInfo)
   at Emby.Dlna.ContentDirectory.ControlHandler.HandleBrowse(XmlWriter xmlWriter, IReadOnlyDictionary`2 sparams, String deviceId)
   at Emby.Dlna.Service.BaseControlHandler.CreateControlResponse(ControlRequestInfo requestInfo)
   at Emby.Dlna.Service.BaseControlHandler.ProcessControlRequestInternalAsync(ControlRequest request)
   at Emby.Dlna.Service.BaseControlHandler.ProcessControlRequestAsync(ControlRequest request)```

**Comment by majid-vaghari** [comment]
@SandyRodgers-2017 Do you have any idea what could cause this problem?

**Comment by SandyRodgers-2017** [comment]
> @SandyRodgers-2017 Do you have any idea what could cause this problem?

Nope. Not even a little bit. It is why I made the issue thread. My issues were fixed by 10.8.1.  It seems that every few updates DLNA breaks. I would suggest downgrading if you can to a release that worked previously for you, if it isn't too old of course. Right now I am on 10.8.3 and it works well enough.

**Comment by 1dolla** [comment]
Wait... What is this log error doing here? Is the DLNA plugin re-scanning media for itself rather than using the metadata already in the Jellyfin database?
